lib/wda/webdriveragent.js

```diff
@@ -44,6 +44,8 @@ class WebDriverAgent {
 
     this.useXctestrunFile = args.useXctestrunFile;
 
+    this.derivedDataPath = args.derivedDataPath;
```

Comment:
Since we don't use `this.derivedDataPath` is only used here, perhaps it's not needed? Just pass in `args.derivedDataPath` into the `XcodeBuild` constructor?